### PR TITLE
#97 Fixed deserializing of EventBridge object body

### PR DESCRIPTION
**Pull request by careri**
*Issue #97*

*Description of changes:* Deserialize EventBridge body if JsonValueKind.Object

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by normj — CHANGES_REQUESTED** [review]
If you could add a unit test for this scenario that would be great.

**Comment by careri** [comment]
Hi, unit test added. (The unit test fails with the old code).

File: test/AWS.Messaging.UnitTests/SerializationTests/EnvelopeSerializerTests.cs [hunk 1]
```diff
@@ -564,6 +564,24 @@ public async Task ConvertToEnvelopeAsync_DataMessageLogging_WithError(bool dataM
...
...8\\u0022,\\u0022time\\u0022:\\u00222024-02-19T09:36:28Z\\u0022,\\u0022region\\u0022:\\u0022eu-central-1\\u0022,\\u0022resources\\u0022:[],\\u0022detail\\u0022:{\\u0022id\\u0022:\\u00226ef9516a-6565-4e81-8177-be89d207fab0\\u0022,\\u0022source\\u0022:\\u0022ServiceNow/Integration\\u0022,\\u0022specversion\\u0022:\\u00221.0\\u0022,\\u0022type\\u0022:\\u0022ChatMessage\\u0022,\\u0022time\\u0022:\\u00222024-02-19T09:36:28.6090473\\u002B00:00\\u0022,\\u0022data\\u0022:\\u0022{\\\\\\u0022MessageDescription\\\\\\u0022:\\\\\\u0022Hello\\\\\\u0022}\\u0022}}\",\"MD5OfBody\":\"fa0ed402839bcce842343be5a5b0e758\",\"MD5OfMessageAttributes\":null,\"MessageAttributes\":{},\"MessageId\":\"83b53c15-8838-4ea1-89ae-97990548f8e4\",\"ReceiptHandle\":\"AQEBrLb6AkV/lUdrvNnEQjxK24LJLPU9\\u002BJko30VRmXIXljTrbfzHGZP3wuGCFSnaBUvUPTHGWU6rxRXCDnifFswSzuCyYjtJV95N3UajyEjkau\\u002BsBZdpYvWE4SKJSrh69HzKCS1QZ8Izula1jDA4dSYDl578kYY67uCKCvyRPtCKq7r5mveThv\\u002BRik1K9zmMrX9urwg8OgCsWcPEOu7duKJEsCSchorNoXjCVRycK7/mqyP1ndeiWc6SBefR9Coiynp2tsA6qFmkCwZNx4hqLINclkV4KyidX5cpdORMY74eoZ/puE6xrwQTJCJL2QKmnegxf645RL0i1tS2nSCeDUVwuQeebaSHbI0dKTyhtFvK/O2V8S39/siAHKz\\u002BCCaUo0YhtJ6FG1tm0xytwo5la2ysi5BKRrlf9thZ16pNaiNmKyI=\"}";
```
Comment (philasmar) [inline test/AWS.Messaging.UnitTests/SerializationTests/EnvelopeSerializerTests.cs:571]: Would you please replace any sensitive info with dummy data? I'm particularly worried about "SenderId".

**Review by philasmar — CHANGES_REQUESTED** [review]

**Review by philasmar — COMMENTED** [review]
I ran the unit tests and integ tests internally and they pass for the current changes.

**Review by 96malhar — COMMENTED** [review]
We have fixed this issue via this PR - https://github.com/awslabs/aws-dotnet-messaging/pull/114
It also adds end-to-end tests for EventBridge and SNS.

This PR can be closed.

**Pull request closed without merging**